app/models/batch_context.rb

```diff
@@ -77,7 +78,21 @@ def bundle_dir_with_path(rel_path)
   # On first call, loads the manifest data, caches results
   # @return [Array<ActiveSupport::HashWithIndifferentAccess>]
   def manifest_rows
-    @manifest_rows ||= CsvImporter.parse_to_hash(bundle_dir_with_path(manifest))
+    @manifest_rows ||= load_manifest
+  end
+
+  # load the manifest of objects and verify there is at least one object and the correct header is present
```

Comment:
is this the "build" manifest or the "file manifest" ?   I find the distinction confusing and wonder if the method names / var names could help

Reply:
yes, it's confusing...this particular one is the main "build" manifest

hopefully your in progress PR should still be ok (except for perhaps modifying the string that shows the error message)

the changes in this PR will throw a job into the failed state if there is a problem with the manifest itself in a more consistent way (the current code catches some situations with the manifest but not all and does it in a weird way)

Reply:
maybe say `manifest.csv` in the comments to be clear?